msal_extensions/cache_lock.py

```diff
@@ -22,6 +22,7 @@ def __enter__(self):
         self._fh.write('{} {}'.format(pid, sys.argv[0]).encode('utf-8'))
 
     def __exit__(self, *args):
+        portalocker.unlock(self._fh)
         self._fh.close()
         try:
             # Attempt to delete the lockfile. In either of the failure cases enumerated below, it is
```

Comment:
I'm quoting @bgavrilMS 's [comment from another PR](https://github.com/AzureAD/microsoft-authentication-extensions-for-python/pull/44#issuecomment-610326593), and answer it here, because the locking functionality is actually provided here.

> ... it is fairly important to delete the file if the process which created it crashes. Otherwise, you will block all other consumers and the only "way out" is for the user to manually delete the lock file. 

> Should not be difficult to implement, in Java this is done like [this](https://github.com/AzureAD/microsoft-authentication-extensions-for-java/blob/master/src/main/java/com/microsoft/aad/msal4jextensions/CrossProcessCacheFileLock.java#L46) and in .net like [this](https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/blob/bac27e967509ee0c37037f2d4ebf9cd51d31d168/src/Shared/CrossPlatLock.cs#L52)

That "DeleteOnClose" behavior provided by in C# and Java is indeed nice. I kind of remember that there was an equivalent thing in the low-level C API (but I [could not find it now](https://www.gnu.org/software/libc/manual/html_node/Opening-and-Closing-Files.html)). Regardless, it is unfortunately that [Python's built-in open() function](https://docs.python.org/3/library/functions.html#open) does not provide this functionality. So the next best thing we can do, is to explicitly remove it at the line below, and live with the potential exception.

Reply:
I suppose a workaround for not having this functionality is to use good old "try" / "finally" in the objects that use the lock. Similar to the "Disposable" concept in C#. I believe you already do that.

Reply:
Check out this SO question, the answer with ~21 votes shows how to get the "DeleteOnClose" functionality https://stackoverflow.com/questions/54772244/whats-a-good-way-to-detect-when-a-python-program-exits-or-crashes

It seems to be different on Win / Linux

Reply:
But that SO q&a were about locking, which is a topic that we happened to already choose its ~39 votes answer for (and I personally even tend to agree with that answer's second part "use a database if at all possible"). :-)

Very few (if any at all) locking library/conversation out there would cover file deletion behavior, simply because file deletion is not part of the usual locking mechanism, which in turn is based on OS's `fcntl.LOCK_EX` on a file handler, not based on application's detection of the existence of a file. It is just our MSAL EX protocol happened to choose to do file deletion (but why?), so I guess we have to implement that by ourselves.

All in all, using an explicit `os.remove(...)` and handling its exception properly, seems adequate.